src/test/java/spoon/test/api/APITest.java

```diff
@@ -639,4 +639,36 @@ public void testProcessModelsTwice() {
 		}
 	}
 
+	@Test
+	public void testParsingInlineQualifiedClassName() {
+		String code1 = "package io.example.pack1.pack2;\n" +
+				"    public class Example{\n" +
+				"      void add(io.example.other.Class1<io.example.other.Class2> value){\n" +
+				"      }\n" +
+				"    }\n";
+		CtClass<?> class1 = Launcher.parseClass(code1);
+		List<CtParameter<?>> add = class1.getMethodsByName("add").get(0).getParameters();
+		assertEquals(1, add.size());
+		assertEquals("value", add.get(0).getSimpleName());
+		assertEquals("io.example.other.Class1", add.get(0).getType().getQualifiedName());
+		assertEquals(1, add.get(0).getType().getActualTypeArguments().size());
+		assertEquals("io.example.other.Class2", add.get(0).getType().getActualTypeArguments().get(0).getQualifiedName());
+		String code2 = "package io.example.pack1.pack2;\n" +
+				"    public class Example{\n" +
+				"      void add(io.example.pack1.Class1<io.example.pack1.Class2> value1, Class1<Class2> value2){\n" +
+				"      }\n" +
+				"    }\n";
+		CtClass<?> class2 = Launcher.parseClass(code2);
+		List<CtParameter<?>> add2 = class2.getMethodsByName("add").get(0).getParameters();
+		assertEquals(2, add2.size());
+		assertEquals("value1", add2.get(0).getSimpleName());
+		assertEquals("io.example.pack1.Class1", add2.get(0).getType().getQualifiedName());
+		assertEquals(1, add2.get(0).getType().getActualTypeArguments().size());
+		assertEquals("io.example.pack1.Class2", add2.get(0).getType().getActualTypeArguments().get(0).getQualifiedName());
+		assertEquals("value2", add2.get(1).getSimpleName());
+		assertEquals("io.example.pack1.pack2.Class1", add2.get(1).getType().getQualifiedName());
+		assertEquals(1, add2.get(1).getType().getActualTypeArguments().size());
+		assertEquals("io.example.pack1.pack2.Class2", add2.get(1).getType().getActualTypeArguments().get(0).getQualifiedName());
```

Comment:
it would be cool if you could use assertj, e.g. `assertThat(add).hasSize(1)` and so on. We also have a `SpoonAssertions.assertThat`, which you can use with spoon things: `assertThat(add.get(0)).getSimpleName().isEqualTo("value")`. You can also refactor this a bit by extracting `add.get(0)` into its own variable, which will likely make this more readable :)